CMakePresets.json

```diff
@@ -1,169 +1,144 @@
-            "name": "linux-amd64",
-            "inherits": [
-                "default"
-            ],
-            "generator": "Ninja Multi-Config",
-            "cacheVariables": {
-                "VCPKG_TARGET_TRIPLET": "x64-linux-heroic",
-                "WITH_HTTP_CURL": "ON",
-                "WITH_WS_WSLAY": "ON",
-                "CFG_CURL_SYSTEM": "ON"
-            },
-            "environment": {
-                "VCPKG_FORCE_SYSTEM_BINARIES": "1"
-            },
-            "condition": {
-                "type": "equals",
-                "lhs": "${hostSystemName}",
-                "rhs": "Linux"
-            }
-        },
-        {
-            "name": "ubuntu1804-amd64",
-            "inherits": [
-                "linux-amd64"
-            ],
-            "hidden": true,
-            "environment": {
-                "CC": "/usr/bin/gcc-11",
-                "CXX": "/usr/bin/g++-11"
-            },
-            "installDir": "out/ubuntu1804-amd64"
-        }
-    ]
+    {
+      "name": "win-x64",
+      "inherits": ["default"],
+      "generator": "Ninja Multi-Config",
+      "cacheVariables": {
+        "CMAKE_SYSTEM_PROCESSOR": "AMD64",
+        "VCPKG_TARGET_TRIPLET": "x64-windows-static-heroic",
+        "VCPKG_CHAINLOAD_TOOLCHAIN_FILE": "${sourceDir}/cmake/toolchains/vcpkg-msvc.cmake",
+        "WITH_WS_LIBHTTPC": "ON",
+        "WITH_HTTP_LIBHTTPC": "ON",
+        "BUILD_TESTING": "ON"
+      },
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Windows"
+      }
+    },
+    {
+      "name": "win-arm64",
+      "inherits": ["default", "win-x64"],
+      "cacheVariables": {
+        "CMAKE_SYSTEM_PROCESSOR": "ARM64",
+        "VCPKG_TARGET_TRIPLET": "arm64-windows-static-heroic"
+      }
+    },
+    {
+      "name": "macosx-universal",
+      "inherits": ["default"],
+      "generator": "Xcode",
+      "cacheVariables": {
+        "VCPKG_TARGET_TRIPLET": "universal-osx-heroic",
+        "CMAKE_OSX_ARCHITECTURES": "arm64;x86_64",
+        "WITH_WS_WSLAY": "ON",
+        "WITH_HTTP_CURL": "ON",
+        "BUILD_TESTING": "ON"
+      },
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Darwin"
+      }
+    },
+    {
+      "name": "ios-arm64",
+      "inherits": ["macosx-universal"],
+      "cacheVariables": {
+        "VCPKG_TARGET_TRIPLET": "arm64-ios-heroic",
+        "CMAKE_SYSTEM_NAME": "iOS",
+        "CMAKE_OSX_ARCHITECTURES": "arm64"
+      }
+    },
+    {
+      "name": "iphonesimulator-arm64",
+      "inherits": ["ios-arm64"],
+      "cacheVariables": {
+        "VCPKG_TARGET_TRIPLET": "arm64-iphonesimulator-heroic",
+        "CMAKE_OSX_SYSROOT": "iphonesimulator"
+      }
+    },
+    {
+      "name": "android-default",
+      "inherits": ["default"],
+      "hidden": true,
+      "generator": "Ninja Multi-Config",
+      "cacheVariables": {
+        "ANDROID_USE_LEGACY_TOOLCHAIN_FILE": "FALSE",
```

Comment:
We should set "ANDROID_USE_LEGACY_TOOLCHAIN_FILE": "FALSE", to true or remove this line.  I don't think it's a good default, if some android platform requires it, we can override it specifically.

Reply:
These presets are not functional: they are to load project in CMAke and expect extra env vars to be set (like NDK path). IDK what to do about them ,delete or fix with some sensible defaults?

Reply:
I would just set to true so our purpose is explicit. If anyone's tool makes any use of it, at least will take sensible defaults. But if it's not functional, don't waste time on it, I'll swap to approved once it finishes my local test run.